builder/builder-next/exporter/writer.go

```diff
@@ -22,11 +22,10 @@ import (
 
 func emptyImageConfig() ([]byte, error) {
 	pl := platforms.Normalize(platforms.DefaultSpec())
-	img := ocispec.Image{
-		Architecture: pl.Architecture,
-		OS:           pl.OS,
-		Variant:      pl.Variant,
-	}
+	img := ocispec.Image{}
+	img.Architecture = pl.Architecture
+	img.OS = pl.OS
+	img.Variant = pl.Variant
```

Comment:
Similar to https://github.com/moby/buildkit/pull/3104/files#r967629557;

This is where I noticed that in some cases we're discarding the `OSVersion` and `OSFeatures` fields (there may be other locations where we do this). Perhaps this is intentional, but we should probably review the code to see if we _unintentionally_ discard those fields, and update the code where needed.

Reply:
FWIW; Variant was added in https://github.com/moby/moby/commit/a2aaf4cc83aa88aa09bb1d4bc88f36bf68cccd66 (https://github.com/moby/moby/pull/43239). Before that, only `OS` and `Architecture` were set (and not normalised)